server/src/main/java/org/elasticsearch/action/admin/indices/close/CloseIndexRequest.java

```diff
@@ -114,4 +118,22 @@ public void writeTo(StreamOutput out) throws IOException {
         indicesOptions.writeIndicesOptions(out);
         waitForActiveShards.writeTo(out);
     }
+
+    public static CloseIndexRequest fromRestRequest(RestRequest request) throws IOException {
+        String[] indices = request.hasParam("index")
+            ? Strings.splitStringByCommaToArray(request.param("index"))
+            : CloseIndexRequest.getIndicesFromBody(request);
+
+        return new CloseIndexRequest(indices);
+    }
+
+    private static String[] getIndicesFromBody(final RestRequest request) throws IOException {
```

Comment:
Generally the `action` layer shouldn't know about REST things. Could we keep the parsing in the REST layer please?

Reply:
Ok, the parsing is being executed in RestCloseIndexAction now.